enhancements/microshift/microshift-router-configuration-security.md

```diff
@@ -0,0 +1,266 @@
+
+Certificate revocation list (CRL) handling was introduced in the router starting from version [4.14](https://github.com/openshift/router/pull/472).
+#### Day 2 Replacing/renewing certificates
+applying the new secret using  kustomize yaml  will be enough for renewing certificate on a running system with minimal downtime
+- using built-in k8s functionality the new secret will be transparently  mounted into the router container
+	 > *When a volume contains data from a Secret, and that Secret is updated, Kubernetes tracks this and updates the data in the volume, using an eventually-consistent approach*.  ([source](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-files-from-a-pod)   )
+- openshift router will detect the updated certificate  and gracefully reload the haproxy ([source](https://github.com/openshift/router/blob/master/pkg/router/template/router.go#L453-L454))
+
+#### Customizing TLSSecurity - Ciphers
+MicroShift currently generates an Ingress deployment with TLS ciphers that support both TLS 1.2 and TLS 1.3, following [**Intermediate**](https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28recommended.29) compatibility guidelines.
+
+Similar to OpenShift Ingress Operators, we want to allow users to select their own TLS Security Profile, offering the following options: **Old, Intermediate, Modern, and Custom**.
+
+The Ingress deployment is controlled by three environment variables:
+
+- **ROUTER_CIPHER** – Specifies ciphers compatible with TLS versions earlier than 1.3.
+- **ROUTER_CIPHERSUITES** – Defines ciphers used exclusively for TLS 1.3 sockets.
+- **SSL_MIN_VERSION** - Minimum allowed SSL version.
+
+If a user selects the **Modern** TLS profile, TLS 1.2 ciphers will be disabled, and only TLS 1.3 will be active with its recommended ciphers, leaving `ROUTER_CIPHERS` empty.
+For the **Custom** TLS profile, users must provide a complete list of supported ciphers. MicroShift will determine whether each cipher should apply to TLS 1.3 or earlier versions.
+
+
+#### How config options change manifests
+Each of the configuration options described above has a direct effect on the
+manifests that MicroShift will apply after starting.  
+see the full Implementation details in the [router-configuration](microshift-router-configuration.md) enhancement.
+
+see the [table](#proposal) in the proposal above for all the new configuration options.
+
+
+
+### Risks and Mitigations
+* User provide certificate and it expired after some time.
+  ingress will continue serving with an expired cert - similiar approach is taken by OpenShift.
+
+- Incorrect ingress configuration can cause network  disruption and unpredictable behavior , the documentation should warn about dangers of changing the defaults. 
+
+### Drawbacks
+
+- External certificate wont be automaticly renewed, so it requires manual Certificate rotation.
```

Comment:
Can we configure any notification in microshift when a cert is about to expire?

Reply:
Similar to the API Server Custom Certificate, external certificate management should be handled by the user. Managing them is outside the scope of this feature.